### PR TITLE
Add support for mac m1

### PR DESCRIPTION
**Pull request by normanmaurer**
Motivation:

As more and more people switch to a mac m1 we should support it

Modifications:

- Add profiles for cross-compile for mac m1
- Adjust script to finish release

Result:

Mac m1 is supported

### CODE REVIEW

File: resolver-dns-native-macos/pom.xml [hunk 1]
```diff
@@ -132,6 +132,111 @@
...
+          </plugin>
+
+          <plugin>
+            <groupId>org.fusesource.hawtjni</groupId>
+            <artifactId>maven-hawtjni-plugin</artifactId>
+            <executions>
+              <execution>
+                <id>build-native-lib</id>
+                <configuration>
+                  <name>netty_resolver_dns_native_macos_${os.detected.arch}</name>
```
Comment (chrisvest) [inline resolver-dns-native-macos/pom.xml:177]: should this be `${os.detected.arch}` for cross-compilation?

Comment (chrisvest) [reply resolver-dns-native-macos/pom.xml:177]: In the kqueue module this is hard coded to `aarch_64`.

**Review by chrisvest — CHANGES_REQUESTED** [review]

**Review by chrisvest — APPROVED** [review]

**Comment by YunWZ** [comment]
[ERROR] Failed to execute goal on project netty-handler: Could not resolve dependencies for project io.netty:netty-handler:jar:5.0.0.Final-SNAPSHOT: io.netty:netty-tcnative:jar:osx-aarch_64:2.0.46.Final was not found in https://repo1.maven.org/maven2 during a previous attempt. This failure was cached in the local repository and resolution is not reattempted until the update interval of Central1 has elapsed or updates are forced -> [Help 1]

**Comment by chrisvest** [comment]
@YunWZ open new issues instead of adding comments to long-closed PRs.

**Comment by YunWZ** [comment]
> @YunWZ open new issues instead of adding comments to long-closed PRs.

I hava commited a new issue #11941

**Comment by YunWZ** [comment]
@chrisvest I have commited a new issue #11941

**Pull request merged**